components/service/telemetry/src/main/java/org/mozilla/telemetry/schedule/jobscheduler/JobSchedulerTelemetryScheduler.java

```diff
@@ -34,4 +35,8 @@ public void scheduleUpload(TelemetryConfiguration configuration) {
 
         scheduler.schedule(jobInfo);
     }
+
+    public void setJobId(int jobId) {
+        this.jobId = jobId;
+    }
```

Comment:
As this ID is only meant to be set once I wonder whether this could be a constructor parameter (secondray, we still need to keep the default constructor with no parameters). Too bad that this isn't Kotlin. :)